pallets/parachain-system/src/relay_state_snapshot.rs

```diff
@@ -235,6 +237,17 @@ impl RelayChainStateProof {
 			.map_err(Error::Config)
 	}
 
+	/// Read parachain [head data](`relay_chain::HeadData`) from the relay chain state proof.
+	///
+	/// Returns an error if anything failed at reading or decoding.
+	pub fn read_para_head(&self) -> Result<Option<relay_chain::HeadData>, Error> {
```

Comment:
please rename this function to something more descriptive e.g. `read_included_para_head` and adjust docs